### PR TITLE
306 : Product Catalogue

### PR DESCRIPTION
**Pull request by rehammuzzamil**
https://github.com/OpenSRP/opensrp-server-core/issues/306

### CODE REVIEW

**Comment by rehammuzzamil** [comment]
There is an issue with the **PUT** end-point. As per the business, the image can also be updated in an edit-mode. Therefore, we need to provide support for the update of both JSON payload and Multipart file in an edit. To make it work, I tried with **@RequestPart** but it did not work. The problem is PUT request represents a single resource. But multi parts means multiple resources in a single body. Therefore, it throws an exception ` Unable to process parts as no multi-part configuration has been provided.`
The solution could be to implement a customized **MultipartResolver** class and override the existing default multipart resolver in an XML configuration to allow a multipart request for both POST and PUT end-point.
Please provide your feedback @dubdabasoduba. Thanks!

File: src/main/java/org/opensrp/repository/postgres/ProductCatalogueRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+		}
+
+		Long uniqueId = retrievePrimaryKey(entity);
+		if (uniqueId == null) {
+			return;
+		}
+
+		ProductCatalogueExample productCatalogueExample = new ProductCatalogueExample();
+		productCatalogueExample.createCriteria().andUniqueIdEqualTo(uniqueId);
+		//		int rowsAffected = customProductCatalogueMapper.deleteByExample(productCatalogueExample);
```
Comment (dubdabasoduba) [inline src/main/java/org/opensrp/repository/postgres/ProductCatalogueRepositoryImpl.java:89]: Please remove this unused piece of code

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

**Comment by dubdabasoduba** [comment]
> There is an issue with the **PUT** end-point. As per the business, the image can also be updated in an edit-mode. Therefore, we need to provide support for the update of both JSON payload and Multipart file in an edit. To make it work, I tried with **@RequestPart** but it did not work. The problem is PUT request represents a single resource. But multi parts means multiple resources in a single body. Therefore, it throws an exception ` Unable to process parts as no multi-part configuration has been provided.`
> The solution could be to implement a customized **MultipartResolver** class and override the existing default multipart resolver in an XML configuration to allow a multipart request for both POST and PUT end-point.
> Please provide your feedback @dubdabasoduba. Thanks!

@rehammuzzamil I agree. I also found the same method after doing some research. So let's work with this solution.

File: src/main/java/org/opensrp/repository/postgres/ProductCatalogueRepositoryImpl.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/repository/postgres/ProductCatalogueRepositoryImpl.java]: Done

**Review by dubdabasoduba — APPROVED** [review]

**Comment by rehammuzzamil** [comment]
As discussed with @dubdabasoduba , closing this PR.

**Pull request closed without merging**